src/main/AuthenticationRequestRegistration.ts

```diff
@@ -11,22 +18,35 @@ export function assertValidRequestRegistrationOpts(opts: SIOP.RequestRegistratio
   }
 }
 
-export function createRequestRegistrationPayload(opts: SIOP.RequestRegistrationOpts): RequestRegistrationPayload {
+export async function createRequestRegistrationPayload(opts: SIOP.RequestRegistrationOpts): Promise<RequestRegistrationPayload> {
   assertValidRequestRegistrationOpts(opts);
+
+  const regObj: SIOP.RPRegistrationMetadataPayload = createRPRegistrationMetadataPayload(opts);
+
+  let regObjToValidate;
+  if (opts.registrationBy.referenceUri) {
+    regObjToValidate = (await getWithUrl(opts.registrationBy.referenceUri)) as unknown as RPRegistrationMetadataPayload;
```

Comment:
I need more context. We can call.

Reply:
Since this is building the request and since we will not have support for creating the byRef payloads initially we cannot call this method yet.
Then there is the problem that even if we had the support to do it, we would need to do it later in the process, after we have ensured we can properly create the request to begin with. So it doesn't make much sense yet to try to validate something which isn't supported yet.

Reply:
fixed. in upcoming commit.